src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1305,6 +1305,12 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
                 {
                     bodyValue = await httpContext.Request.ReadFromJsonAsync(jsonTypeInfo);
                 }
+                catch (BadHttpRequestException ex) when (ex.StatusCode == StatusCodes.Status413RequestEntityTooLarge)
```

Comment:
I understand wanting to be conservative in servicing, but this seems overkill. We trust the exceptions thrown by the server or we don't, that shouldn't vary by status code.

Reply:
> We trust the exceptions thrown by the server or we don't, that shouldn't vary by status code.

This is a valid point. I was definitely cautious about changing the response behavior for a large swath of exception scenarios but it's valid that RDF shouldn't be setting status codes differently from what Kestrel is doing.

Reply:
Updated based on feedback. Minor thought on style ... thoughts on doing something like this in the existing catch block:

```csharp
httpContext.Response.StatusCode = ex is BadHttpRequestException bad ? bad.StatusCode : StatusCodes.Status400BadRequest;
```

My instinct is no -- even though the code is a bit repetitive I think its clearer.